### PR TITLE
Fix geographic bounding box coordinate validation and aggregation precision

### LINKED ISSUE #11559: The validateBoundingBox method fails with values having more than 5 decimal places.
**What steps does it take to reproduce the issue?**

Create a Dataset with a Geographic Bounding Box field.

Enter coordinates with more than 5 decimal places - for example, set South greather than North or West greather than East using 6 decimal places.

* When does this issue occur?

Geographic Bounding Box validation.

* Which page(s) does it occurs on?

Create/Edit Dataset

* What happens?

When a user creates a Geographic Bounding Box field and enters values with more than 5 decimal places, the validation is not correct. Dataverse saves the Dataset, but Solr cannot index the item and throws an exception.

* To whom does it occur (all users, curators, superusers)?

All users

* What did you expect to happen?

Dataverse thows error message: "Geographic Bounding Box has invalid coordinates. East must be greater than West and North must be greater than South. Missing values are NOT allowed."

**Which version of Dataverse are you using?**

First observed in version 6.4, and it still occurs in 6.6.

**Any related open or closed issues to this bug report?**

No

**Screenshots:**

No matter the issue, screenshots are always welcome.

To add a screenshot, please use one of the following formats and/or methods described here:

Image

Image

**Are you thinking about creating a pull request for this issue?**
Help is always welcome, is this bug something you or your organization plan to fix?

Yes.

The method verifyBoundingBoxCoordinatesWithinRange should compare BigDecimal values instead of Float for greater precision. This does not affect performance if used only to validate coordinates.

### PR DESCRIPTION
**Pull request by tuannx**
**What this PR does / why we need it**:
- Adds a REST Assured integration test (`testGeographicBoundingBoxHighPrecisionValidation` in `DatasetsIT`) exercising the bounding box precision bug #11559.
- Fixes single-precision float truncation in `DatasetFieldValueValidator` by parsing and comparing bounding box coordinates with `BigDecimal`.
- Updates bounding box aggregation in `IndexServiceBean` to use `BigDecimal`.
- Adds unit tests in `DatasetFieldValueValidatorTest` covering high-precision edge cases.
- Adds regression unit test in `IndexServiceBeanTest` verifying high-precision multi-box aggregation.
- Adds release note snippet in `doc/release-notes/11559-bbox-bigdecimal.md`.

**Which issue(s) this PR closes**:
- Closes #11559

**Special notes for your reviewer**:
This PR includes the reproduction test in `DatasetsIT` and the `BigDecimal` precision fix in `DatasetFieldValueValidator` / `IndexServiceBean`.

Minor cleanups included:
- Modernized legacy `assertTrue(!...)` assertions to `assertFalse(...)` in `DatasetFieldValueValidatorTest`.
- Added defensive coordinate whitespace trimming and ensured only valid bounding boxes are aggregated in `IndexServiceBean`.

**Suggestions on how to test this**:
- Unit tests: `mvn test -Dtest=DatasetFieldValueValidatorTest,IndexServiceBeanTest`
- Integration test: `mvn test -Dtest=DatasetsIT#testGeographicBoundingBoxHighPrecisionValidation`

**Does this PR introduce a user interface change?**: No
**Is there a release notes update needed for this change?**: Yes (included)

### CODE REVIEW

**Comment by pdurbin** [comment]
@tuannx thanks for the PR! We'll discuss it at Triage Tuesday and get back to you! 😄

**Comment by tuannx** [comment]
[ERROR] Failures: 
[ERROR]   DatasetsIT.testGeographicBoundingBoxHighPrecisionValidation:7555 1 expectation failed.
Expected status code (<400> or <403>) but was <201>.

Solr index:

Remote error message: ERROR: [doc=dataset_4_draft] Error adding field 'geolocation'='ENVELOPE(-71.0,-70.0,42.001000,42.001001)' 
msg=Unable to parse shape given formats "lat,lon", "x y" or as WKT because 
org.locationtech.spatial4j.exception.InvalidShapeException: maxY must be >= minY: 42.001001 to 42.001

File: doc/release-notes/11559-bbox-bigdecimal.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## Bug ##
+
+Fixed an issue where geographic bounding box coordinates with more than 5 decimal places could pass validation despite reversed order (due to single-precision float precision loss) and subsequently cause Solr indexing failures with `InvalidShapeException`. Bounding box coordinates are now parsed and compared using `BigDecimal`.
```
Comment (landreev) [inline doc/release-notes/11559-bbox-bigdecimal.md:3]: I am debating whether it may make sense to suggest to selectively reindex datasets with bounding boxes in the metadata, in case any of them were not properly indexed because of the bug. 
I would prefer not to tell _everybody_ to run a full reindex because of this fix; because that is a time-consuming affair for large instances, and the bug is somewhat exotic. 

Should this note say something like "At the time of the release, review the combined release note and the upgrade instructions. If we are not recommending a full reindex for other reasons, consider adding a suggestion to selective reindex datasets with bounding boxes" - ? 

I don't have a strong opinion on this, just a thought.

Comment (landreev) [reply doc/release-notes/11559-bbox-bigdecimal.md:3]: Other than this, looks great, happy to approve and move forward. 
Thank you for making the PR!

**Review by landreev — APPROVED** [review]
@tuannx I just realized I had a write permission on your fork. Added that one piece to the release note; approving the PR.

**Comment by tuannx** [comment]
> @tuannx I just realized I had a write permission on your fork. Added that one piece to the release note; approving the PR.

Thank @landreev to update release not and approve the PR.

File: src/test/java/edu/harvard/iq/dataverse/DatasetFieldValueValidatorTest.java [hunk 2]
```diff
@@ -174,12 +174,33 @@ public void testBoundingBoxValidity() {
...
+        assertFalse(DatasetFieldValueValidator.validateBoundingBox("-180", "180", "90", "-90.0000001"));
+
+        // Still valid with many decimals when order is correct
+        assertTrue(DatasetFieldValueValidator.validateBoundingBox("-71.116431", "-71.116430", "42.377001", "42.377000"));
+        // High-precision point bounding box (min == max)
+        assertTrue(DatasetFieldValueValidator.validateBoundingBox("-71.116431", "-71.116431", "42.377000", "42.377000"));
+        // Leading/trailing whitespace should be trimmed and accepted
+        assertTrue(DatasetFieldValueValidator.validateBoundingBox(" -71.116431 ", " -71.116430 ", " 42.377001 ", " 42.377000 "));
     }
 }
```
Comment (pdurbin) [inline src/test/java/edu/harvard/iq/dataverse/DatasetFieldValueValidatorTest.java:206]: I'm just noting here at the bottom that I clicked the button to approve workflows (to make sure tests are passing):

Image
